src/script/common/c_content.cpp

```diff
@@ -1959,8 +1959,28 @@ void push_objectRef(lua_State *L, const u16 id)
 
 void read_hud_element(lua_State *L, HudElement *elem)
 {
-	elem->type = (HudElementType)getenumfield(L, 2, "hud_elem_type",
-									es_HudElementType, HUD_ELEM_TEXT);
+	std::string type_string;
+	bool has_type = getstringfield(L, 2, "type", type_string);
+
+	// Handle deprecated hud_elem_type
+	std::string deprecated_type_string;
+	if (getstringfield(L, 2, "hud_elem_type", deprecated_type_string)) {
+		if (has_type && deprecated_type_string != type_string) {
+			script_log_unique(L, "Ambiguous HUD element fields \"type\" and \"hud_elem_type\", "
```

Comment:
any reason to not make this an error?

Reply:
At first I used an error but later changed it to a warning, since it can break backward comparability in some very cases.
Imagine a deprecated mod doing this:
```lua
...
id = player:hud_add({hud_elem_type = "image", ...})
...
-- Change hud_elem_type from "image" to "text"
local def = player:hud_get(id)
player:hud_remove(id)
def.hud_elem_type = "text" -- Needed because hud_get returns a type and not a hud_elem_type field
id = player:hud_add(def) -- Results in ambiguous HUD element fields
```